cwms-data-api/src/test/java/cwms/cda/api/BasinControllerTestIT.java

```diff
@@ -1,75 +1,514 @@
+				.body("basin-id[0].office-id", equalTo(BASIN.getBasinId().getOfficeId()))
+				.body("[0].sort-order", equalTo(BASIN.getSortOrder().floatValue()))
+				.body("[0].area-unit", equalTo(BASIN.getAreaUnit()))
+				.body("[0].total-drainage-area", equalTo(BASIN.getTotalDrainageArea().floatValue()))
+				.body("[0].contributing-drainage-area", equalTo(BASIN.getContributingDrainageArea().floatValue()))
+				.body("parent-basin-id[0].name", equalTo(BASIN.getParentBasinId().getName()))
+				.body("parent-basin-id[0].office-id", equalTo(BASIN.getParentBasinId().getOfficeId()))
+			;
+		} else {
+			// Retrieve basin and assert that it exists
+			given()
+				.log().ifValidationFails(LogDetail.ALL, true)
+				.accept(Formats.JSONV1)
+				.queryParam(Controllers.OFFICE, OFFICE)
+			.when()
+				.redirects().follow(true)
+				.redirects().max(3)
+				.get("basins/" + BASIN.getBasinId().getName())
+			.then()
+				.log().ifValidationFails(LogDetail.ALL, true)
+			.assertThat()
+				.statusCode(is(HttpServletResponse.SC_OK))
+				.body("basin-id[0].name", equalTo(BASIN.getBasinId().getName()))
+				.body("basin-id[0].office-id", equalTo(BASIN.getBasinId().getOfficeId()))
+				.body("[0].sort-order", equalTo(BASIN.getSortOrder().floatValue()))
+				.body("[0].area-unit", equalTo(BASIN.getAreaUnit()))
+				.body("[0].total-drainage-area", equalTo(BASIN.getTotalDrainageArea().floatValue()))
+				.body("[0].contributing-drainage-area", equalTo(BASIN.getContributingDrainageArea().floatValue()))
+			;
+		}
+
+		// Delete basin
+		given()
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(Formats.JSONV1)
+			.queryParam(Controllers.OFFICE, OFFICE)
+			.queryParam(DELETE_MODE, DELETE_ACTION)
+			.header(AUTH_HEADER, user.toHeaderValue())
+		.when()
+			.redirects().follow(true)
+			.redirects().max(3)
+			.delete("basins/" + BASIN.getBasinId().getName())
+		.then()
+			.log().ifValidationFails(LogDetail.ALL, true)
+		.assertThat()
+			.statusCode(is(HttpServletResponse.SC_NO_CONTENT))
+		;
+
+		// Retrieve basin and assert that it does not exist
+		given()
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(Formats.JSONV1)
+			.queryParam(Controllers.OFFICE, OFFICE)
+		.when()
+			.redirects().follow(true)
+			.redirects().max(3)
+			.get("basins/" + BASIN.getBasinId().getName())
+		.then()
+			.log().ifValidationFails(LogDetail.ALL, true)
+		.assertThat()
+			.statusCode(is(HttpServletResponse.SC_NOT_FOUND))
+		;
 	}
 
-	@EnumSource(AliasTest.class)
-	@ParameterizedTest
-	void test_getAll_aliases(AliasTest test)
-	{
+	@Test
+	void test_update_does_not_exist() {
+		TestAccounts.KeyUser user = TestAccounts.KeyUser.SWT_NORMAL;
+		// Create basin
 		given()
-			.log().ifValidationFails(LogDetail.ALL,true)
-			.accept(test._accept)
-			.queryParam(Controllers.OFFICE, SPK)
+			.log().ifValidationFails(LogDetail.ALL, true)
+			.accept(Formats.JSONV1)
+			.contentType(Formats.JSONV1)
+			.queryParam(Controllers.OFFICE, OFFICE)
```

Comment:
update this to not use query params, matching your update to the controller

Reply:
Fixed in latest push